### PR TITLE
update verifier interface

### LINKED ISSUE #71: task: [ingestor] update verifier interface with Key wrapper
Update the Verifier interface that adds in the Key Wrapper and register new providers.

### PR DESCRIPTION
**Pull request by pxp928**
Signed-off-by: pxp928 <parth.psu@gmail.com>

Verifier interface updated with helper functions and unit tests

Closes #71

### CODE REVIEW

**Review by mlieberman85 — APPROVED** [review]
LGTM

**Comment by pxp928** [comment]
This will have to rebase slightly once the in-memory key provider PR #78 merges. Small change to the `keyutil` functions that is being used by the unit tests here.

**Comment by pxp928** [comment]
unit test fixed based on change to the `keyutil` function. Ready for merge!

File: pkg/ingestor/verifier/verifier.go [hunk 1]
```diff
@@ -42,6 +49,29 @@ type Verifier interface {
...
 }
+
+var (
+	verifierProviders = map[VerifierType]Verifier{}
+)
+
+// RegisterVerifier registers the providers that are available for verification
+func RegisterVerifier(k Verifier, providerType VerifierType) {
+	if _, ok := verifierProviders[providerType]; ok {
+		logrus.Warnf("the verification provider is being overwritten: %s", providerType)
```
Comment (lumjjb) [inline pkg/ingestor/verifier/verifier.go:63]: we changed this in the logger PR to just return an error instead.

File: pkg/ingestor/verifier/verifier_test.go [hunk 2]
```diff
@@ -0,0 +1,164 @@
...
+		wantErr: true,
+	}}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			got, err := VerifyIdentity(tt.doc)
+			if (err != nil) != tt.wantErr {
+				t.Errorf("VerifyIdentity() error = %v, wantErr %v", err, tt.wantErr)
+				return
+			}
+			if !reflect.DeepEqual(got, tt.want) {
```
Comment (lumjjb) [inline pkg/ingestor/verifier/verifier_test.go:159]: if its an error case, it should not check this

**Review by lumjjb — APPROVED** [review]
minor changes if not lgtm

File: pkg/ingestor/verifier/verifier.go (hunk 1 above)
Comment (pxp928) [reply pkg/ingestor/verifier/verifier.go:63]: yup. changed.

File: pkg/ingestor/verifier/verifier_test.go (hunk 2 above)
Comment (pxp928) [reply pkg/ingestor/verifier/verifier_test.go:159]: done.

**Pull request merged**